### ISSUE TITLE
Cannot find -lGL

### ISSUE THREAD

**Issue by shashaanktulsyan** [gluonhq/client-maven-plugin#127]
I am getting the following error 

`gcc /home/user/client-samples-gluon-mvn/Maven/HelloFX/target/client/x86_64-linux/gvm/hellofx/launcher.o /home/user/client-samples-gluon-mvn/Maven/HelloFX/target/client/x86_64-linux/gvm/tmp/SVM-1582218774277/hellofx.hellofx.o -Wl,-Bstatic -ljava -lnio -lzip -lnet -lprefs -ljvm -lstrictmath -lj2pkcs11 -lsunec -lextnet -llibchelper -Wl,-Bdynamic -lz -ldl -lstdc++ -lpthread -rdynamic -Wl,--whole-archive -lprism_es2 -lglass -lglassgtk3 -ljavafx_font -ljavafx_font_freetype -ljavafx_font_pango -ljavafx_iio -Wl,--no-whole-archive -lGL -lX11 -lgtk-3 -lgdk-3 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lfreetype -lpangoft2-1.0 -lgthread-2.0 -lstdc++ -lz -lXtst -o /home/user/client-samples-gluon-mvn/Maven/HelloFX/target/client/x86_64-linux/hellofx -L/home/user/Development/graalvm/graalvm-ce-java11-20.0.0/lib/svm/clibraries/linux-amd64 -L/home/user/Development/graalvm/graalvm-ce-java11-20.0.0/lib -L/home/user/.gluon/substrate/javafxStaticSdk/14-ea+gvm6/linux-x86_64/sdk/lib
/usr/bin/x86_64-linux-gnu-ld: cannot find -lGL`

I was able to javafx:compile and client:link to work following the steps given gluon client page.

A few days ago I tried this on windows using Windows subsystem for linux - ubuntu 16.0.4, that gave this same error. That time I tried the following and none of this worked. 

- `sudo apt-get install libgl`    did not work
- `sudo apt-get install libgl1-mesa-glx`    did not work
-  `sudo apt-get install cmake libx11-dev xorg-dev libglu1-mesa-dev freeglut3-dev libglew1.5 libglew1.5-dev libglu1-mesa libglu1-mesa-dev libgl1-mesa-glx libgl1-mesa-dev libglfw3-dev libglfw3` did not work; taking reference from some comment in - https://medium.com/@theorose49/install-opengl-at-ubuntu-18-04-lts-31f368d0b14e & https://github.com/gluonhq/client-gradle-plugin/issues/57 
- `sudo apt-get install libglfw3-dev libgles2-mesa-dev
gcc glfw_triangle.c -lGLESv2 -lglfw` this also not worked

Using the above steps I was able to run the link maven target, but the resultant binary would not run saying it cannot find a suitable pipeline, neither es2 nor sw. 
I have very old laptop upgraded over time. The graphics card is now not supported. But if javafx is able to run on java as it is, it should be able to run after compilation from graalvm also, is my logic. And at any case it should be able to fallback to sw.

This time i tried using a virtual machine with kubuntu 16.0.4 and I am getting the same error. 
I just don't want to install incorrect libraries and make things messy. So before I try different solutions I wanted to take advice/help/tips from experts. 

Once this works, will switch to the other custom graal version which builds for android. Right now testing for linux first.

Thank you

**Comment by jperedadnr** [comment]
You can add
```
<enableSWRendering>true</enableSWRendering>
```
to the plugin. Then try again?

**Comment by shashaanktulsyan** [comment]
> You can add
> 
> ```
> <enableSWRendering>true</enableSWRendering>
> ```
> 
> to the plugin. Then try again?

So I tried what I understood and it did not work. 
I have 2 questions
- Where to put this ``<enableSWRendering>true</enableSWRendering>`` ? 
```
    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.source>11</maven.compiler.source>
        <maven.compiler.target>11</maven.compiler.target>
        <javafx.version>13.0.1</javafx.version>
        <client.plugin.version>0.1.14</client.plugin.version>
        <mainClassName>hellofx.HelloFX</mainClassName>
	<enableSWRendering>true</enableSWRendering>
    </properties>
```

- **After putting this flag, do I need to just run the ``mvn client:link`` or ``mvn client:compile`` also ** ? 

In the meantime I am trying different permutations of this to see if it works.

Thanks

**Comment by jperedadnr** [comment]
Add "to the plugin" means:

```
           <plugin>
                <groupId>com.gluonhq</groupId>
                <artifactId>client-maven-plugin</artifactId>
                <version>0.1.15</version>
                <configuration>
                      ...
                      <enableSWRendering>true</enableSWRendering>
                </configuration>
            </plugin>
```

**Comment by shashaanktulsyan** [comment]
> Add "to the plugin" means:
> 
> ```
>            <plugin>
>                 <groupId>com.gluonhq</groupId>
>                 <artifactId>client-maven-plugin</artifactId>
>                 <version>0.1.15</version>
>                 <configuration>
>                       ...
>                       <enableSWRendering>true</enableSWRendering>
>                 </configuration>
>             </plugin>
> ```

Thank you it worked, recompiled and linked, it took a lot of time, but it worked. Around 60MB build for HelloFX sample. 


While it was compiling I recollected that last time when I tried this, although I compiled it on this old laptop, I tried to run it in a more recent hardware which had a ubuntu and centos both, supporting opengl es 3 ( whereas javafx's requirements are met with version 2 itself if I am not wrong). Even in that case it did not run on either, giving the error that it could not find suitable pipeline. It gave other errors also, like it complained of some libraries being missing etc. I guess I will look into that when I have time. If I face this issue again, I will create another ticket.

Thank you again.
